src/hostname/hostnamed.c

```diff
@@ -250,8 +250,27 @@ static int get_firmware_vendor(char **ret) {
          return get_hardware_firmware_data("bios_vendor", ret);
 }
 
-static int get_firmware_date(char **ret) {
-         return get_hardware_firmware_data("bios_date", ret);
+static int get_firmware_date(uint64_t *ret) {
+         const char *k;
+         struct tm tm;
+         time_t v;
+         _cleanup_free_ char *bios_date = NULL;
+         int r;
+
+         r = get_hardware_firmware_data("bios_date", &bios_date);
+         if (r <= 0)
+                 return r;
+
+         k = strptime(bios_date, "%m/%d/%Y", &tm);
+         if (!k || *k)
+                 return r;
+
+        v = timegm(&tm);
+        if (v == (time_t) -1)
+                return r;
+
+        *ret = (uint64_t) v * USEC_PER_SEC;
```

Comment:
maybe add extra safety overflow check here first:

```c
if (v > UINT64_MAX / USEC_PER_SEC) /* overflow check */
        return -EINVAL;
```

just in case somebody specifies some ridiculously large year there.